code/modules/mob/living/carbon/xenomorph/Castes/Queen.dm

```diff
@@ -391,17 +364,18 @@
 	var/queensWord = "<br><h2 class='alert'>The words of the queen reverberate in your head...</h2>"
 	queensWord += "<br><span class='alert'>[input]</span><br>"
 
+	INVOKE_ASYNC(src, .proc/do_hive_message, queensWord)
+
+/mob/living/carbon/Xenomorph/Queen/proc/do_hive_message(queensWord)
 	if(SSticker?.mode)
-		for(var/datum/mind/L in SSticker.mode.xenomorphs)
-			var/mob/living/carbon/Xenomorph/X = L.current
-			if(X && X.client && istype(X) && !X.stat && hivenumber == X.hivenumber)
-				X << sound(get_sfx("queen"),wait = 0,volume = 50)
-				to_chat(X, "[queensWord]")
+		hive.xeno_message("[queensWord]")
+		for(var/i in hive.get_watchable_xenos())
+			var/mob/living/carbon/Xenomorph/X = i
+			X << sound(get_sfx("queen"),wait = 0,volume = 50)
 
-	spawn(0)
-		for(var/mob/dead/observer/G in GLOB.player_list)
-			G << sound(get_sfx("queen"),wait = 0,volume = 50)
-			to_chat(G, "[queensWord]")
+	for(var/mob/dead/observer/G in GLOB.player_list)
```

Comment:
```suggestion
	for(var/mob/dead/observer/G in GLOB.observer_list)
```

Reply:
With that you can also replace the typecheck

Reply:
good find